src/main/resources/runner/docker/haskell/Dockerfile

```diff
@@ -1,12 +1,16 @@
 FROM haskell:8
-# TODO FIND A SMALLER BASE IMAGE TO RUN ON ALPINE
-# TODO ADD ANY OTHER DEPENDENCIES
 
 ARG DIR
+ARG USER=appuser
 WORKDIR $DIR
-USER root
+RUN adduser $USER
 
-ADD . .
-RUN chmod a+x haskell_entry.sh
+RUN chown -R $USER $DIR
+
+USER $USER
 
-ENTRYPOINT ["sh", "haskell_entry.sh"]
+RUN stack install HUnit

```

Comment:
Can we move this instruction up to after `FROM haskell:8` since it's so large to optimize layers?

Reply:
I am assuming you are talking about the `RUN stack...` instruction. I say it is best to leave it after the `USER $USER` instruction since it installs the package locally, for the user. So calling it earlier will install `HUnit` for root and will need to be reinstalled for `$USER`.

It comes before `ADD . .`, so it shouldn't need to be rerun for every test case. So, the store image will not be small, but every run will take a significantly short time to build the image. If I understood you correctly.

Reply:
Yeah I meant' the `stack install` instruction. I didn't know we would need to re-install it for the user if it was already installed for root. Don't worry about moving it then.